src/internal/operators/take.ts

```diff
@@ -51,20 +51,41 @@ export function take<T>(count: number): MonoTypeOperatorFunction<T> {
     : operate((source, subscriber) => {
         let seen = 0;
         source.subscribe(
-          new OperatorSubscriber(subscriber, (value) => {
-            // Increment the number of values we have seen,
-            // then check it against the allowed count to see
-            // if we are still letting values through.
-            if (++seen <= count) {
-              subscriber.next(value);
-              // If we have met or passed our allowed count,
-              // we need to complete. We have to do <= here,
-              // because re-entrant code will increment `seen` twice.
-              if (count <= seen) {
+          new OperatorSubscriber(
+            subscriber,
+            (value) => {
+              // Increment the number of values we have seen,
+              // then check it against the allowed count to see
+              // if we are still letting values through.
+              if (++seen <= count) {
+                subscriber.next(value);
+                // If we have met or passed our allowed count,
+                // we need to complete. We have to do <= here,
+                // because re-entrant code will increment `seen` twice.
+                if (count <= seen) {
+                  subscriber.complete();
+                }
+              }
+            },
+            () => {
+              // If seen === count or higher, then we've already taken all of
+              // the values we were supposed to, and the complete we're getting here
+              // is from reentrant code racing our `complete` above. We want to stop
+              // that here.
+              if (seen < count) {
                 subscriber.complete();
               }
+            },
+            (err) => {
+              // If seen === count or higher, then we've already taken all of
+              // the values we were supposed to, and the error we're getting here
+              // is from reentrant code racing our `complete` above. We want to stop
+              // that here.
+              if (seen < count) {
+                subscriber.error(err);
+              }
```

Comment:
out of interest: is there any particular reason, why blocking notifications over unsubscribing before the last next notification was used?

(besides the failing delayWhen test)

Reply:
Sorry for the late response. I'm not sure I understand your question here.